.ci/releasenotes-gen-auto.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -e
+
+echo "PULL_REQUEST:""$PULL_REQUEST"
+if [[ $PULL_REQUEST =~ ^([0-9]+)$ ]]; then
+  echo "Build is not for Pull Request";
+  sleep 5;
+  exit 0;
+fi
+
+mkdir -p .ci-temp
+if [ -d .ci-temp/contribution ]; then
+  cd .ci-temp/contribution/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/contribution
+  cd ../
+fi
+cd .ci-temp/contribution/releasenotes-builder
+mvn -e --no-transfer-progress clean compile package
+cd ../../../
+
+# we need to do full clone as Travis do "git clone --depth=50"
+if [ -d .ci-temp/checkstyle ]; then
+  cd .ci-temp/checkstyle/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/checkstyle
+  cd ../
+fi
+cd .ci-temp/checkstyle
+LATEST_RELEASE_TAG=$(git describe $(git rev-list --tags --max-count=1))
+cd ../../
+
+CS_RELEASE_VERSION=$(mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
+              -Dexec.args='${project.version}' \
+              --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec | sed 's/-SNAPSHOT//')
+echo LATEST_RELEASE_TAG="$LATEST_RELEASE_TAG"
+echo CS_RELEASE_VERSION="$CS_RELEASE_VERSION"
+cd .ci-temp
+java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.jar \
+        -localRepoPath checkstyle -remoteRepoPath checkstyle/checkstyle \
+        -startRef "$LATEST_RELEASE_TAG" -releaseNumber "$CS_RELEASE_VERSION" \
+        -githubAuthToken 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 \
+        -generateXdoc \
+        -xdocTemplate contribution/releasenotes-builder/src/main/resources/com/github/checkstyle/\
+/templates/xdoc_freemarker.template \
+        -publishXdoc -publishXdocWithPush
+
```

Comment:
Use variable to shorten a path to template

Reply:
Changes Done

Reply:
I have given the RW access (755), but `checkchmod` in `drone.io` is still failing.